core/capabilities/syncer.go

```diff
@@ -37,65 +53,99 @@ func NewRegistrySyncer(peerWrapper p2ptypes.PeerWrapper, registry types.Capabili
 		"12D3KooWJbZLiMuGeKw78s3LM5TNgBTJHcF39DraxLu14bucG9RN",
 		"12D3KooWGqfSPhHKmQycfhRjgUDE2vg9YWZN27Eue8idb2ZUk6EH",
 	}
-	peers := make(map[ragetypes.PeerID]p2ptypes.StreamConfig)
-	donInfo := &remotetypes.DON{
-		ID: "don1",
+	capabilityDONPeers := []string{
+		"12D3KooWHCcyTPmYFB1ydNvNcXw5WyAomRzGSFu1B7hpB4yi8Smf",
+		"12D3KooWPv6eqJvYz7TcQWk4Y4XjZ1uQ7mUKahdDXj65ht95zH6a",
 	}
-	for _, peerID := range peerIDs {
-		var p ragetypes.PeerID
-		err := p.UnmarshalText([]byte(peerID))
-		if err != nil {
-			return err
+	allPeers := make(map[ragetypes.PeerID]p2ptypes.StreamConfig)
+	process := func(peers []string, donInfo *remotetypes.DON) error {
+		for _, peerID := range peers {
+			var p ragetypes.PeerID
+			err := p.UnmarshalText([]byte(peerID))
+			if err != nil {
+				return err
+			}
+			allPeers[p] = defaultStreamConfig
+			donInfo.Members = append(donInfo.Members, p)
 		}
-		peers[p] = defaultStreamConfig
-		donInfo.Members = append(donInfo.Members, p)
+		return nil
+	}
+	workflowDonInfo := remotetypes.DON{ID: "workflowDon1"}
+	if err := process(workflowDONPeers, &workflowDonInfo); err != nil {
+		return err
 	}
-	err := s.peerWrapper.GetPeer().UpdateConnections(peers)
+	capabilityDonInfo := remotetypes.DON{ID: "capabilityDon1"}
+	if err := process(capabilityDONPeers, &capabilityDonInfo); err != nil {
+		return err
+	}
+	err := s.peerWrapper.GetPeer().UpdateConnections(allPeers)
 	if err != nil {
 		return err
 	}
 	// NOTE: temporary hard-coded capabilities
-	capId := "sample_remote_target"
-	targetCap := remote.NewRemoteTargetCaller(commoncap.CapabilityInfo{
+	capId := "sample_remote_trigger"
+	triggerInfo := commoncap.CapabilityInfo{
 		ID:             capId,
-		CapabilityType: commoncap.CapabilityTypeTarget,
-		Description:    "Remote Target",
+		CapabilityType: commoncap.CapabilityTypeTrigger,
+		Description:    "Remote Trigger",
 		Version:        "0.0.1",
-	}, donInfo, s.dispatcher, s.lggr)
-	err = s.registry.Add(ctx, targetCap)
-	if err != nil {
-		s.lggr.Error("failed to add remote target capability to registry")
-		return err
 	}
-	err = s.dispatcher.SetReceiver(capId, donInfo.ID, targetCap)
+	myId := s.peerWrapper.GetPeer().ID().String()
+	config := remotetypes.RemoteTriggerConfig{
+		RegistrationRefreshMs: 20000,
+	}
+	var srv services.Service
+	if slices.Contains(workflowDONPeers, myId) {
+		s.lggr.Info("member of a workflow DON - starting remote subscribers")
+		triggerCap := remote.NewTriggerSubscriber(config, triggerInfo, capabilityDonInfo, workflowDonInfo, s.dispatcher, nil, s.lggr)
+		err = s.registry.Add(ctx, triggerCap)
+		if err != nil {
+			s.lggr.Errorw("failed to add remote target capability to registry", "error", err)
+			return err
+		}
+		err = s.dispatcher.SetReceiver(capId, capabilityDonInfo.ID, triggerCap)
+		if err != nil {
+			s.lggr.Errorw("failed to set receiver", "capabilityId", capId, "donId", capabilityDonInfo.ID, "error", err)
+			return err
+		}
+		srv = triggerCap
+	} else {
```

Comment:
There's technically two more possible cases here, isn't there? (i.e. neither a member of a capability or a workflow DON, and a member of both).

Reply:
True. I'll change to "if a member of capability DON". Again, this is only temporary to support hardcoded test values.